FiveCalls/FiveCalls/CallScriptViewController.swift

```diff
@@ -35,48 +31,27 @@ class CallScriptViewController : UIViewController, IssueShareable {
         super.viewDidLoad()
 
         navigationController?.navigationBar.tintColor = .white
-        
         navigationItem.rightBarButtonItem = UIBarButtonItem(barButtonSystemItem: .action, target: self, action: #selector(IssueDetailViewController.shareButtonPressed(_ :)))
         
         tableView.estimatedRowHeight = 100
         tableView.rowHeight = UITableViewAutomaticDimension
-        
-        setupView()
     }
     
-    func setupView() {
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
         guard let issue = issue, let issueIndex = issue.contacts.index(where:{$0.id == contact.id}) else { return }
         title = "Contact \(issueIndex+1) of \(issue.contacts.count)"
-
-        let hasCompleted = logs.hasCompleted(issue: issue.id, allContacts: issue.contacts)
-        let hasContacted = logs.hasContacted(contactId: contact.id, forIssue: issue.id)
-        if hasCompleted {
-            self.footerHeightContraint.constant = 30
-            self.footerLabel.text = "You've contacted everyone, great work!"
-        } else {
-            self.footerHeightContraint.constant = hasContacted ? 75 : 118
-            self.footerLabel.text = hasContacted ? "You've already contacted \(self.contact.name)." : "Enter your call result to get the next call."
-        }
-        self.resultNextButton.isHidden = hasCompleted || !hasContacted
-        self.resultUnavailableButton.isHidden = hasCompleted || hasContacted
-        self.resultVoicemailButton.isHidden = hasCompleted || hasContacted
-        self.resultContactedButton.isHidden = hasCompleted || hasContacted
-        self.resultSkipButton.isHidden = hasCompleted || hasContacted
-        
-        footer.setNeedsUpdateConstraints()
     }
     
     func callButtonPressed(_ button: UIButton) {
         callNumber(contact.phone)
     }
 
     func callNumber(_ number: String) {
-        print("dialing \(number)")
+        lastPhoneDialed = number
         if let dialURL = URL(string: "telprompt:\(number)") {
             if #available(iOS 10.0, *) {
-                UIApplication.shared.open(dialURL) { success in
-                    //Log the result
-                }
+                UIApplication.shared.open(dialURL)
             } else {
                 UIApplication.shared.openURL(dialURL)
```

Comment:
We have this check in two places. Mind moving to a category on UIApplication, something like:

```swift
UIApplication.shared.fvc_open(_ url:...)
```

?